File: hyperLiquid/src/types/order.ts
export interface OrderRecord {
  partitionKey: string;
  id: string;
  user_address: string;
  symbol: string;
  strategy: string;
  quantity: number;
  order_type: string;
  price: number;
  oid: string;
  stopLossOid?: string;
  stopLossPrice?: number;
  pnl?: number;
  status: string;
  created_at: Date;
  updated_at: Date;
}

export interface NewOrder {
  user_address: string;
  symbol: string;
  strategy: string;
  quantity: number;
  order_type: string;
  price: number | string;
  oid: string;
  stopLossOid?: string;
  stopLossPrice?: number | string;
  status: string;
}

File: src/hyperliquid_python/models/order.py
from pydantic import BaseModel
from typing import Optional, Union
from datetime import datetime

class OrderRecord(BaseModel):
    partitionKey: str
    id: str
    user_address: str
    symbol: str
    strategy: str
    quantity: float
    order_type: str
    price: float
    oid: str
    stopLossOid: Optional[str] = None
    stopLossPrice: Optional[float] = None
    pnl: Optional[float] = None
    status: str
    created_at: datetime
    updated_at: datetime

class NewOrder(BaseModel):
    user_address: str
    symbol: str
    strategy: str
    quantity: float
    order_type: str
    price: Union[float, str]
    oid: str
    stopLossOid: Optional[str] = None
    stopLossPrice: Optional[Union[float, str]] = None
    status: str
